Core/Src/main.c

```diff
@@ -105,15 +105,13 @@ int main(void)
   {
 	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET); //setting the PB4 port (I think that's the GPIO port?) low
 
-	  HAL_SPI_TransmitReceive(&hspi1, T_buffer, R_buffer, 8, 500);
-	  adc_value = (R_buffer[1] << 8) | R_buffer[2];
+	  HAL_SPI_TransmitReceive(&hspi1, T_buffer, R_buffer, 3, 500);
+	  adc_value = ((R_buffer[1] << 8) | R_buffer[2]) && 0x3FF;
```

Comment:
&& is the logical operator, you are are looking for & which is the bitwise operator!